src/syntax/FStar.Syntax.Embeddings.fst

```diff
@@ -84,9 +84,43 @@ let mk_emb_full em un typ printer emb_typ = {
     emb_typ = emb_typ
 }
 
+//
+// AR/NS: 04/22/2022:
+//        In the case of metaprograms, we reduce divergent terms in
+//        the normalizer, therefore, the final result that we get
+//        may be wrapped in a Meta_monadic node (e.g. lift, app, etc.)
+//        Before unembedding the result of such a computation,
+//          we strip those meta nodes
+//        In case the term inside is not a result, unembedding would
+//          anyway fail
+//        And we strip down only DIV
+//        Can we get any other effect? Not today, since from the client
+//          code, we enforce terms to be normalized to be PURE
+//
+
+let rec unmeta t =
```

Comment:
A nit, but it may be clearer to call this function something more explicit, like `unmeta_div_results` or something like that. We already and `unmeta` in FStar.Syntax.Util, which we've considered dangerous to use in the past, with variants like unmeta_safe etc. I worry about future bugs where by mistakenly opening FStar.Syntax.Util we end up silently calling the unsafe unmeta rather than just stripping Meta_monadic div and lifts to div on results.

Reply:
Will change, thanks!